rs/types/types/src/messages/inter_canister.rs

```diff
@@ -653,6 +655,57 @@ impl From<Response> for RequestOrResponse {
     }
 }
 
+/// A message that acts as a blocker for the message behind it in the `StreamIndexedQueue` of a
+/// `StreamSlice` until the `begin` in a different `StreamSlice` coming from `subnet_id` is verified
+/// to be >= `index`.
```

Comment:
I'd rather put this in terms of streams. We're blocking the stream until some other stream has advanced past an index.

Stream slices are just the mechanism by which we pull streams from other subnets and what we put into a payload. And `StreamIndexedQueue` is very much an implementation detail; the random reader of this comment will have no idea about it.

Instead I would add a bit of detail (as in, one sentence) as to what this is used for: to prevent out-of-order delivery of messages due to races between streams after canister migrations.